rust-src/id/src/ffi.rs

```diff
@@ -292,6 +299,274 @@ pub extern "C" fn ip_info_cdi_verify_key(
     ptr
 }
 
+/// Attempt to create an [`ArInfo`] instance from byte array pointers.
+/// Assumes that public_key_ptr points to a PublicKey<G1>
```

Comment:
I don't see that this function assumes that public key points to that. Please update documentation.